### PR TITLE
importer: Check whether file exist before importing

### PR DESCRIPTION
**Pull request by Little-Wallace**
### What problem does this PR solve?

Issue Number: close https://github.com/tikv/tikv/issues/9496

Problem Summary:
If TiKV has applied one command of ingest but tidb-lightning(import-client) thinks this request has failed, it would send the same ingest-request to TiKV again, which may cause TiKV panic because the file has been ingested and could not be found.
There is many case which may cause request failed while it has been applied. Such as:
1. Network lost some packets of ingest-response.
2. The command had been copied to more than half of the members of raft-group but the original leader crashed before the entry of this command committed, then the new leader committed this entry but the client has received  fail messages by the response from the original leader.

### What is changed and how it works?

We can propose one ingest-request when all the following conditions are met:
1. There is not any pending ingest-request for this sst.
2. The current leader has applied to current term, and we can ensure that the file which we want to ingest exist.
3. The ingest file must be removed before we return response to client, so that the new request will never see the same file again.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - It may increase the duration of ingest request.
- Breaking backward compatibility

### Release note 

- No release note.

### CODE REVIEW

**Review by kennytm — COMMENTED** [review]
Rest LGTM

**Comment by Little-Wallace** [comment]
@kennytm PTAL

**Comment by kennytm** [comment]
```
[2021-01-21T08:03:39.223Z] test import::test_sst_service::test_ingest_sst ... thread 'main' panicked at 'assertion failed: resp.has_error()', tests/integrations/import/test_sst_service.rs:105:5
[2021-01-21T08:03:39.223Z] stack backtrace:
[2021-01-21T08:03:39.479Z]    0: std::panicking::begin_panic
[2021-01-21T08:03:39.479Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/std/src/panicking.rs:505:12
[2021-01-21T08:03:39.479Z]    1: integrations::import::test_sst_service::test_ingest_sst
[2021-01-21T08:03:39.479Z]              at ./tests/integrations/import/test_sst_service.rs:105:5
[2021-01-21T08:03:39.479Z]    2: integrations::import::test_sst_service::test_ingest_sst::{{closure}}
[2021-01-21T08:03:39.479Z]              at ./tests/integrations/import/test_sst_service.rs:89:1
[2021-01-21T08:03:39.479Z]    3: core::ops::function::FnOnce::call_once
[2021-01-21T08:03:39.479Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/core/src/ops/function.rs:227:5
[2021-01-21T08:03:39.479Z]    4: test_util::runner::run_test_with_hook::{{closure}}::{{closure}}
[2021-01-21T08:03:39.479Z]              at ./components/test_util/src/runner.rs:64:21
[2021-01-21T08:03:39.479Z]    5: core::ops::function::FnOnce::call_once{{vtable.shim}}
[2021-01-21T08:03:39.479Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/core/src/ops/function.rs:227:5
[2021-01-21T08:03:39.479Z]    6: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
[2021-01-21T08:03:39.479Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/alloc/src/boxed.rs:1042:9
[2021-01-21T08:03:39.479Z] note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

File: src/import/sst_service.rs [hunk 1]
```diff
@@ -239,58 +257,98 @@ where
...
                 }
-                Err(e) => Err(e),
+
+                // Make ingest command.
+                let mut ingest = Request::default();
+                ingest.set_cmd_type(CmdType::IngestSst);
+                ingest.mut_ingest_sst().set_sst(m.clone());
+
+                let mut cmd = RaftCmdRequest::default();
+                cmd.set_header(header);
```
Comment (gengliqi) [inline src/import/sst_service.rs:314]: Should set `term` to avoid leader change between read and write

**Review by gengliqi — APPROVED** [review]
LGTM

**Comment by kennytm** [comment]
https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/tikv_ghpr_test/runs/40874/nodes/107/log/?start=0

Test failure seems to be #7092.

<details><summary>

```
[2021-01-21T11:48:47.868Z] test cases::test_merge::test_node_merge_recover_snapshot ... thread 'main' panicked at 'request timeout', /home/jenkins/agent/tikv-master/build/components/test_raftstore/src/cluster.rs:778:9
[2021-01-21T11:48:47.868Z] stack backtrace:
[2021-01-21T11:48:47.868Z]    0: std::panicking::begin_panic
[2021-01-21T11:48:47.868Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/std/src/panicking.rs:505:12
[2021-01-21T11:48:47.868Z]    1: test_raftstore::cluster::Cluster<T>::request
[2021-01-21T11:48:47.868Z]              at ./components/test_raftstore/src/cluster.rs:778:9
[2021-01-21T11:48:47.868Z]    2: test_raftstore::cluster::Cluster<T>::must_put_cf
[2021-01-21T11:48:47.868Z]              at ./components/test_raftstore/src/cluster.rs:911:20
[2021-01-21T11:48:47.868Z]    3: test_raftstore::cluster::Cluster<T>::must_put
[2021-01-21T11:48:47.868Z]              at ./components/test_raftstore/src/cluster.rs:907:9
[2021-01-21T11:48:47.868Z]    4: failpoints::cases::test_merge::test_node_merge_recover_snapshot
[2021-01-21T11:48:47.868Z]              at ./tests/failpoints/cases/test_merge.rs:404:9
[2021-01-21T11:48:47.868Z]    5: failpoints::cases::test_merge::test_node_merge_recover_snapshot::{{closure}}
[2021-01-21T11:48:47.868Z]              at ./tests/failpoints/cases/test_merge.rs:370:1
[2021-01-21T11:48:47.868Z]    6: core::ops::function::FnOnce::call_once
[2021-01-21T11:48:47.868Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/core/src/ops/function.rs:227:5
[2021-01-21T11:48:47.868Z]    7: test_util::runner::run_test_with_hook::{{closure}}::{{closure}}
[2021-01-21T11:48:47.868Z]              at ./components/test_util/src/runner.rs:64:21
[2021-01-21T11:48:47.868Z]    8: core::ops::function::FnOnce::call_once{{vtable.shim}}
[2021-01-21T11:48:47.868Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/core/src/ops/function.rs:227:5
[2021-01-21T11:48:47.868Z]    9: <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once
[2021-01-21T11:48:47.869Z]              at /rustc/b1496c6e606dd908dd651ac2cce89815e10d7fc5/library/alloc/src/boxed.rs:1042:9
```

</summary></details>

**Pull request merged**